analytical_engine/apps/simple_path/all_simple_paths.h

```diff
@@ -0,0 +1,324 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+Author: Ma JingYuan<nn9902@qq.com>
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_ALL_SIMPLE_PATHS_H_
+#define ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_ALL_SIMPLE_PATHS_H_
+
+#include <set>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "simple_path/all_simple_paths_context.h"
+
+#include "core/app/app_base.h"
+#include "core/worker/default_worker.h"
+
+namespace gs {
+
+template <typename FRAG_T>
+class AllSimplePaths : public AppBase<FRAG_T, AllSimplePathsContext<FRAG_T>>,
+                       public grape::Communicator {
+ public:
+  INSTALL_DEFAULT_WORKER(AllSimplePaths<FRAG_T>, AllSimplePathsContext<FRAG_T>,
+                         FRAG_T)
+  static constexpr grape::MessageStrategy message_strategy =
+      grape::MessageStrategy::kAlongOutgoingEdgeToOuterVertex;
+  static constexpr grape::LoadStrategy load_strategy =
+      grape::LoadStrategy::kBothOutIn;
+  using vertex_t = typename fragment_t::vertex_t;
+  using vid_t = typename fragment_t::vid_t;
```

Comment:
add a `using msg_t = tuple<....>` here and replace the message declare with `msg_t`